tests/unit_tests/fn/test_fn_sid_neighbor_table.cu

```diff
@@ -0,0 +1,65 @@
+/*
+ * GridTools
+ *
+ * Copyright (c) 2014-2021, ETH Zurich
+ * All rights reserved.
+ *
+ * Please, refer to the LICENSE file in the root directory.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include <gridtools/fn/sid_neighbor_table.hpp>
+
+#include <array>
+#include <cstddef>
+#include <cstdint>
+#include <type_traits>
+
+#include <gtest/gtest.h>
+
+#include <cuda_test_helper.hpp>
+#include <gridtools/sid/synthetic.hpp>
+
+namespace gridtools::fn {
+    namespace {
+        using sid_neighbor_table::as_neighbor_table;
+
+        using edge_dim_t = integral_constant<int_t, 0>;
+        using edge_to_cell_dim_t = integral_constant<int_t, 1>;
+
+        template <class Table>
+        __device__ auto neighbor_table_neighbors_device(Table const &table, int index)
+            -> array<std::int32_t, 2> {
```

Comment:
```suggestion
        __device__ auto neighbor_table_neighbors_device(Table const &table, int index) {
```

Reply:
I tried, but:
```
[build] <PATH>/tests/unit_tests/fn/test_fn_sid_neighbor_table.cu(54): error: A non-constexpr __device__ function ("neighbor_table_neighbors_device") with "auto" deduced return type cannot be directly referenced outside the bodies of device functions, except if the reference is absent when __CUDA_ARCH__ is undefined
```
What's weird is that it first builds it, and complains on rebuild.